### PR TITLE
profiles: unmask qt6

### PR DESCRIPTION
**Pull request by Chiitoo**
No known issues that would justify the mask, and the planned split of parts is not something that should block these from being more available-like at this time.

### CODE REVIEW

File: profiles/package.mask [hunk 1]
```diff
@@ -832,31 +832,6 @@ dev-python/shiboken6
 dev-python/pyside6
 dev-python/pyside6-tools
 
-# Jimi Huotari <chiitoo@gentoo.org> (2022-08-02)
```
Comment (thesamesam) [inline profiles/package.mask:835]: Let's include the pyside6 mask above as well.

Comment (thesamesam) [reply profiles/package.mask:835]: And I suppose the games-emulation/pcsx2 mask too..

**Review by thesamesam — APPROVED** [review]
This is fine with me. There's no clear proposals on what any further split should be precisely, no volunteers to do the work, and more and more applications are starting to support (and in some cases require) Qt 6.

Also, we're a bit behind other distros in terms of general Qt 6 availability, so...

**Comment by Chiitoo** [comment]
> This is fine with me. There's no clear proposals on what any further split should be precisely

I think it was mostly (if not only) about splitting qttools, which I did start working on but have not had the time to make any better from the very crude first-try example I did (probably need to create a new package that have the common CMake files or something like that).

File: profiles/package.mask (hunk 1 above)
Comment (ionenwks) [reply profiles/package.mask:835]: Yeah please do, it was only separate because wanted a custom mask message for surprised users.

Comment (ionenwks) [reply profiles/package.mask:835]: Also, what about USE=qt6? Not many packages will be able to use qt6 if not unmasked on arch/amd64/use.mask (I'd leave the base mask on for now).

Flag is typically off by default for now so it shouldn't have a big impact unless people opt-in.

Comment (Chiitoo) [reply profiles/package.mask:835]: Definitely forgot about the flag, yeah, and did not think of the other packages.

Thanks!

**Review by AndrewAmmerlaan — APPROVED** [review]

File: profiles/package.mask [hunk 2]
```diff
@@ -882,39 +882,6 @@ sys-cluster/slurm
 # See: <https://github.com/martijnbastiaan/doctest-parallel/issues/45>
 dev-haskell/doctest-parallel
 
-# Andrew Ammerlaan <andrewammerlaan@gentoo.org> (2022-08-12)
-# Masked for testing, depends on dev-qt/qt*:6
-# Pyside6 is stuck on python3_10 for the moment being
-dev-python/shiboken6
-dev-python/pyside6
-dev-python/pyside6-tools
```
Comment (AndrewAmmerlaan) [inline profiles/package.mask:890]: The ˋpyside6ˋ and ˋpyqt6ˋflag masks on ˋdev-python/QtPyˋ can also be dropped.

**Review by Pesa — COMMENTED** [review]
No objections from me.

**Comment by a17r** [comment]
I'll try to take a look at the binary link name situation today evening.

**Comment by RarogCmex** [comment]
I think qt6 packages will be usable even without splitting.

File: profiles/package.mask (hunk 1 above)
Comment (darkbasic) [reply profiles/package.mask:904]: > I'd leave the base mask on for now

@ionenwks not a fan of this... we're really late on qt6 and it's about time we give it more testing. I can do my part on ppc64le.

Comment (ionenwks) [reply profiles/package.mask:904]: Well, once it's unmasked we can go through normal keywording requests and /then/ unmask it on ppc64le and such.

File: dev-python/pyside6/pyside6-6.4.2.ebuild [hunk 3]
```diff
@@ -87,7 +87,7 @@ RDEPEND="${PYTHON_DEPS}
 		x11-libs/libxkbcommon
 	)
 	help? ( =dev-qt/qttools-${QT_PV}[assistant] )
-	multimedia? ( =dev-qt/qtmultimedia-${QT_PV}[qml(+)?,gles2-only(-)=,widgets(+)?] )
```
Comment (thesamesam) [inline dev-python/pyside6/pyside6-6.4.2.ebuild:90]: I'd `git mv` for this to avoid confusion as I expect a fair number of people have unmasked Qt 6.

**Comment by a17r** [comment]
You could already pick these changes for master and let the unmask stuff remain in this PR exclusively.

**Comment by Chiitoo** [comment]
Should be all done.

Is @rndxelement OK with unmasking of 'app-editors/retext-8.0.0' here?

Also @thesamesam OK with 'media-video/bino-2.0'?

**Comment by rndxelement** [comment]
> Should be all done.
> 
> Is @rndxelement OK with unmasking of 'app-editors/retext-8.0.0' here?
> 
> Also @thesamesam OK with 'media-video/bino-2.0'?
> 

I'm okay with the unmask, thanks!

**Comment by a17r** [comment]
I've begun looking a bit at this in terms of install image and this looks odd:

`/usr/lib64/qt6/bin/qtdiag6`

It should probably be qtdiag6 in `/usr/bin` linking to `/usr/lib64/qt6/bin/qtdiag` instead, right?

**Comment by ionenwks** [comment]
> I've begun looking a bit at this in terms of install image and this in dev-qt/qttools looks odd:
> 
> `/usr/lib64/qt6/bin/qtdiag6`
> 
> It should probably be qtdiag6 in `/usr/bin` linking to `/usr/$(get_libdir)/qt6/bin/qtdiag` instead, right?
> 
> Basically, available links in PATH should mirror the current state of Qt5 ebuilds, just with '6' suffix instead.

Can it generate a similar .txt with "links for packagers" for this one? Generally feel like it should be what Qt6 tells us to create. Referring to https://bugs.gentoo.org/863395#c9 and https://doc.qt.io/qt-6/packaging-recommendations.html (maybe automating based on the .txt would be better, albeit not sure how many there are to know if it's worth it)

**Comment by Chiitoo** [comment]
> Can it generate a similar .txt with "links for packagers"

Yes, indeed.

```
/usr/lib64/qt6/bin/linguist /usr/bin/linguist6
/usr/lib64/qt6/bin/designer /usr/bin/designer6
/usr/lib64/qt6/bin/pixeltool /usr/bin/pixeltool6
/usr/lib64/qt6/bin/assistant /usr/bin/assistant6
/usr/lib64/qt6/bin/qtplugininfo /usr/bin/qtplugininfo6
/usr/lib64/qt6/bin/qdbus /usr/bin/qdbus6
/usr/lib64/qt6/bin/qdbusviewer /usr/bin/qdbusviewer6
/usr/lib64/qt6/bin/qtdiag /usr/bin/qtdiag6
```

**Comment by Chiitoo** [comment]
I guess something like this would work, though could indeed parse the generated text file... and handle it in the eclass but testing this for now.

```
src_install() {
    qt6-build_src_install

    use assistant && dosym ../$(get_libdir)/qt6/bin/linguist /usr/bin/linguist6
    use designer && dosym ../$(get_libdir)/qt6/bin/designer /usr/bin/designer6
    use distancefieldgenerator && dosym ../$(get_libdir)/qt6/bin/qdistancefieldgenerator /usr/bin/qdistancefieldgenerator6
    use linguist && dosym ../$(get_libdir)/qt6/bin/linguist /usr/bin/linguist6
    use pixeltool && dosym ../$(get_libdir)/qt6/bin/pixeltool /usr/bin/pixeltool6
    use qdbus && dosym ../$(get_libdir)/qt6/bin/qdbus /usr/bin/qdbus6
    use qdbus && dosym ../$(get_libdir)/qt6/bin/qdbusviewer /usr/bin/qdbusviewer6
    use qdoc && dosym ../$(get_libdir)/qt6/bin/qdoc /usr/bin/qdoc6
    use qtdiag && dosym ../$(get_libdir)/qt6/bin/qtdiag /usr/bin/qtdiag6
    use qtplugininfo && dosym ../$(get_libdir)/qt6/bin/qtplugininfo /usr/bin/qtplugininfo6
}
```

**Comment by a17r** [comment]
You may also take a look at `qt5_symlink_binary_to_path` from `qt5-build.eclass`.

**Comment by Chiitoo** [comment]
> You may also take a look at `qt5_symlink_binary_to_path` from `qt5-build.eclass`.

Oh yeah!

Will "port" that over with 6.4.3.

**Comment by Chiitoo** [comment]
Pushed 6.4.3 with the symlinks on qttools, and restricted qtwebengine[system-ffmpeg] to <ffmpeg-5 since it seems a modified ffmpeg is expected to be used with it [1] (nothing will be built otherwise, which got me to finally add the check from qtwebengine-5 that makes the build error out if nothing is built, instead of successfully installing some cmake files).

1. https://code.qt.io/cgit/qt/qtwebengine.git/commit/?h=v6.4.3&id=56e1dc813c3067d9fe1fb94f9683a16598c8b8e6

**Comment by Chiitoo** [comment]
And qt6 for all...

**Pull request closed without merging**